File: src/easy/Q1480RunningSumof1dArray.java
package easy;

public class Q1480RunningSumof1dArray {
    public int[] runningSum(int[] nums) {
        int[] out = new int[nums.length];

        out[0] = nums[0];
        for(int i = 1; i < nums.length; i++) {
            out[i] = out[i-1] + nums[i];
        }
        return out;
    }
}

File: src/easy/Q1480RunningSumof1dArray.kt
package easy

fun runningSum(nums: IntArray): IntArray {
    if(nums.size <= 1) {
        return nums;
    }

    for(i in 1 until nums.size) {
        nums[i] = nums[i] + nums[i-1]
    }
    return nums;
}